lib/puppet/file_serving/http_metadata.rb

```diff
@@ -20,7 +20,21 @@ def initialize(http_response, path = '/dev/null')
       checksum = checksum.unpack("m0").first.unpack("H*").first
       @checksums[:md5] = "{md5}#{checksum}"
     end
+    # Add support for artifactory headers.
+    # This adds support for all three algorithms, sha256, sha1, and md5.
+    if checksum = http_response['X-Checksum-Md5']
```

Comment:
These are non-standard, and could potentially be used for a different purpose on other websites. As such, I'm not comfortable hard-coding these headers as fallbacks.

However, this could be done as settings that can be configured in `puppet.conf`. See https://github.com/puppetlabs/puppet/blob/master/lib/puppet/defaults.rb#L825-L839 for an example of a setting that takes an array. My suggestion would be to introduce 3 settings - one each for custom md5, sha1, and sha256 headers - that can be an array of possible headers. Then `#initialize` would iterate through them to look for a valid checksum.